network/src/commonMain/kotlin/com/wire/kalium/network/api/conversation/CreateConversationRequest.kt

```diff
@@ -37,10 +43,22 @@ data class CreateConversationRequest(
 )
 
 
-@Serializable
+@Serializable(with = ReceiptMode.ReceiptModeAsIntSerializer::class)
 enum class ReceiptMode(val value: Int) {
     DISABLED(0),
     ENABLED(1);
+
+    object ReceiptModeAsIntSerializer : KSerializer<ReceiptMode> {
+        override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor("ReceiptMode", PrimitiveKind.INT)
+        override fun serialize(encoder: Encoder, value: ReceiptMode) {
+            encoder.encodeInt(value.value)
+        }
+
+        override fun deserialize(decoder: Decoder): ReceiptMode {
+            val value = decoder.decodeInt()
+            return if (value > 0) ReceiptMode.ENABLED else ReceiptMode.DISABLED
```

Comment:
should it be value > 0 ? it makes me think the value could be anything between 0 ... n but it is a boolean 0,1 ?

Reply:
It can be anything between 0 .. n, and we treat every positive number as true